### PR TITLE
create sumo tarball with leap dependencies (Closes: #6683)

### PR DESCRIPTION
**Pull request by kalikaneko**
NOT to merge yet, there's yet a remaining issue with the _versions for each module that needs to be fixed.

### CODE REVIEW

**Comment by kalikaneko** [comment]
versions fixed
(btw, ivanalejandro, I thik maybe the bundle could use this sumo tarball since it already fixes the modules reported versions, which I think the bundles does not do yet).

**Comment by ivanalejandro0** [comment]
> (btw, ivanalejandro, I thik maybe the bundle could use this sumo tarball since it already fixes the modules reported versions, which I think the bundles does not do yet).

The bundler do get the reported versions in https://github.com/leapcode/bitmask_bundler/blob/develop/pkg/bitmask-0.8.0.json and git-clone/checkout versions as well as specific branches.
I'm most inclined towards the bundler itself doing all the bundling/releasing stuff. Let me think about it.

File: setup.py [hunk 1]
```diff
@@ -240,13 +252,168 @@ def run(self):
 
 
 class cmd_sdist(versioneer_sdist):
+
+    user_options = versioneer_sdist.user_options + \
+        [('sumo', 's',
+          "create a 'sumo' sdist which includes the contents of tahoe-deps/*"),
```
Comment (ivanalejandro0) [inline setup.py]: tahoe-deps?

Comment (kalikaneko) [inline setup.py]: oops... fixed copypasta

**Comment by kalikaneko** [comment]
> The bundler do get the reported versions

No, what I mean is that the bundler gets the right thing, but it doesn't fix the **reported** versions (ie, what's in _version.py), what should do in my opinion. So you end with "unknown" version reported for all the leap* stuff except for bitmask.

Since we're going to do a sumo tarball with all the leap deps for each release, I think it might simplify the bundling for the final releases, since it will be wgetting one tar.gz vs checking out all the repos.

For sure the ability to checkout arbitrary branches for internal, testing bundles should be retained.

**Comment by ivanalejandro0** [comment]
rebased and merged on `develop` branch.

**Pull request closed without merging**